### PR TITLE
0.0.1 - Dev2

### PR DESCRIPTION
**Pull request by SticksDeveloper**
Added a bunch of things and painfully long blockbase replaced resources tab to place it in MainRegistry (Easier)

### CODE REVIEW

**Review by JustDesoroxxx — CHANGES_REQUESTED** [review]

File: src/main/java/cc/panada/mwc/registry/MainRegistry.java [hunk 1]
```diff
@@ -34,11 +47,13 @@ public class MainRegistry {
     // Blocks
 
     public static final List<Block> blocks = new ArrayList<>();
-
+    public static final Block WORKBENCH = new BlockBase("workbench", Material.IRON, 3.0f, 5.0f, "pickaxe", 1, MainRegistry.otherTab,false,false,
+            new AxisAlignedBB(-0.45D, 0.0D, 0.0D, 1.55D, 1.0D, 1.0D), new AxisAlignedBB(0.0D, 0.0D, -0.45D, 1.0D, 1.0D, 1.55D),false);
```
[the remark this answers is not in the document]
Comment (SticksDeveloper) [reply src/main/java/cc/panada/mwc/registry/MainRegistry.java:51]: sorry always do that whenever a line goes over my screen

File: src/main/java/cc/panada/mwc/other/BlockBase.java [hunk 2]
```diff
@@ -3,25 +3,49 @@
...
 public class BlockBase extends Block {
 
+    private final boolean canBlockStayBoolean;
+
+    private boolean canBlockStay(World worldIn, BlockPos pos) {
+        return worldIn.getBlockState(pos.down()).isSideSolid(worldIn, pos, EnumFacing.UP);
+    }
+
+    public static final PropertyDirection FACING = PropertyDirection.create("facing", EnumFacing.Plane.HORIZONTAL);
+    {
```
Comment (JustDesoroxxx) [inline src/main/java/cc/panada/mwc/other/BlockBase.java:31]: Wrong syntax (Brackets)

File: src/main/java/cc/panada/mwc/other/BlockBase.java [hunk 3]
```diff
@@ -30,10 +54,57 @@ public BlockBase(String name, Material soundMaterial, float hardness, float resi
...
+    }
+
+    @Override
+    public void observedNeighborChange(IBlockState state, World worldIn, BlockPos pos, Block blockIn, BlockPos fromPos) {
+        if(!this.canBlockStayBoolean) {
+
+        } else {
+            worldIn.setBlockToAir(pos);
+            InventoryHelper.spawnItemStack(worldIn, pos.getX(), pos.getY(), pos.getZ(), new ItemStack(MainRegistry.WORKBENCH));
+        }
```
Comment (JustDesoroxxx) [inline src/main/java/cc/panada/mwc/other/BlockBase.java:100]: Empty `if` block, remove the `!` and the brackets and just put the code currently under else under if

File: src/main/java/cc/panada/mwc/other/BlockBase.java [hunk 4]
```diff
@@ -45,7 +116,17 @@ public boolean isFullCube(IBlockState state) {
     }
 
     @Override
-    public AxisAlignedBB getBoundingBox(IBlockState state, IBlockAccess source, BlockPos pos) {
-        return BlockModel;
+    public AxisAlignedBB getBoundingBox(IBlockState state, IBlockAccess source, BlockPos pos)
+    {
+        switch(((EnumFacing)state.getValue(FACING)))
+        {
```
Comment (JustDesoroxxx) [inline src/main/java/cc/panada/mwc/other/BlockBase.java:122]: Wrong syntax (Brackets)

File: src/main/java/cc/panada/mwc/other/BlockBase.java [hunk 5]
```diff
@@ -30,10 +54,57 @@ public BlockBase(String name, Material soundMaterial, float hardness, float resi
...
+    }
+
+    @Override
+    protected BlockStateContainer createBlockState() {
+        return new BlockStateContainer(this, new IProperty[] {FACING});
+    }
+
+    @Override
+    public IBlockState getStateForPlacement(World worldIn, BlockPos pos,EnumFacing facing, float hitX, float hitY, float hitZ, int meta, EntityLivingBase placer)
+    {
```
Comment (JustDesoroxxx) [inline src/main/java/cc/panada/mwc/other/BlockBase.java:84]: Wrong syntax (Brackets)

File: src/main/java/cc/panada/mwc/other/BlockBase.java [hunk 6]
```diff
@@ -45,7 +116,17 @@ public boolean isFullCube(IBlockState state) {
...
+        {
+            case SOUTH:
+            default:
+                return BlockModel;
+            case EAST:
+            case WEST:
+                return BlockModel_WE;
+        }
     }
+
```
Comment (JustDesoroxxx) [inline src/main/java/cc/panada/mwc/other/BlockBase.java:131]: Remove whitespace

File: src/main/java/cc/panada/mwc/registry/MainRegistry.java [hunk 7]
```diff
@@ -18,7 +20,18 @@ public class MainRegistry {
 
     // Creative Tabs
 
-    public static final CreativeTabs resourceTab = new ResourcesTab(CreativeTabs.getNextID(), "resourceTab");
+    public static final CreativeTabs resourceTab = new CreativeTabs(CreativeTabs.getNextID(), "resourceTab") {
+        @Override
+        public ItemStack createIcon() {
+            return new ItemStack(MainRegistry.GUNMETAL_INGOT);
+        }
```
Comment (JustDesoroxxx) [inline src/main/java/cc/panada/mwc/registry/MainRegistry.java:27]: This is either one tab or space too forward

File: src/main/java/cc/panada/mwc/registry/MainRegistry.java [hunk 8]
```diff
@@ -34,11 +47,12 @@ public class MainRegistry {
     // Blocks
 
     public static final List<Block> blocks = new ArrayList<>();
-
+    public static final Block WORKBENCH = new BlockBase("workbench", Material.IRON, 3.0f, 5.0f, "pickaxe", 1, MainRegistry.otherTab, false, false, new AxisAlignedBB(-0.45D, 0.0D, 0.0D, 1.55D, 1.0D, 1.0D), new AxisAlignedBB(0.0D, 0.0D, -0.45D, 1.0D, 1.0D, 1.55D),false);
     public static final Block COPPER_ORE = new BlockOreBase("copper_ore", COPPER_CHUNK, 3.0f, 5.0f, "pickaxe", 2, resourceTab);
     public static final Block SULFUR_ORE = new BlockOreBase("sulfur_ore", SULFUR, 2.0F, 5.0F, "pickaxe", 2, resourceTab);
     public static final Block TIN_ORE = new BlockOreBase("tin_ore", TIN, 2.0F, 5.0F, "pickaxe", 2, resourceTab);
 
+
```
Comment (JustDesoroxxx) [inline src/main/java/cc/panada/mwc/registry/MainRegistry.java:55]: Remove double white space

File: src/main/resources/assets/mwc/lang/en_us.lang [hunk 9]
```diff
@@ -35,10 +35,11 @@ menu.example3.description=Enables example feature.
 
 # Creative Tabs
 itemGroup.resourceTab=MWC Resources
+itemGroup.otherTab=MWC Workbench's | Others
```
Comment (JustDesoroxxx) [inline src/main/resources/assets/mwc/lang/en_us.lang:38]: For now should maybe just be called `MWC General`

**Review by JustDesoroxxx — CHANGES_REQUESTED** [review]

File: src/main/java/cc/panada/mwc/other/BlockBase.java (hunk 4 above)
Comment (SticksDeveloper) [reply src/main/java/cc/panada/mwc/other/BlockBase.java:122]: how? is it invalid syntax

Comment (SticksDeveloper) [reply src/main/java/cc/panada/mwc/other/BlockBase.java:122]: oh you mean curly brackets

File: src/main/resources/assets/mwc/lang/en_us.lang (hunk 9 above)
Comment (SticksDeveloper) [reply src/main/resources/assets/mwc/lang/en_us.lang:38]: Got it

**Comment by Paneedah** [comment]
Looks good to me.

File: src/main/java/cc/panada/mwc/registry/MainRegistry.java [hunk 10]
```diff
@@ -34,7 +49,7 @@ public class MainRegistry {
     // Blocks
 
     public static final List<Block> blocks = new ArrayList<>();
-
+    public static final Block WORKBENCH = new BlockBase("workbench", Material.IRON, 3.0f, 5.0f, "pickaxe", 1, MainRegistry.otherTab, false, false, new AxisAlignedBB(-0.45D, 0.0D, 0.0D, 1.55D, 1.0D, 1.0D), new AxisAlignedBB(0.0D, 0.0D, -0.45D, 1.0D, 1.0D, 1.55D),false);
```
Comment (JustDesoroxxx) [inline src/main/java/cc/panada/mwc/registry/MainRegistry.java:52]: every `0.0D` should be just `0` same with every non decimal numbers

**Review by JustDesoroxxx — CHANGES_REQUESTED** [review]

File: src/main/java/cc/panada/mwc/registry/MainRegistry.java [hunk 11]
```diff
@@ -34,7 +49,7 @@ public class MainRegistry {
     // Blocks
 
     public static final List<Block> blocks = new ArrayList<>();
-
+    public static final Block WORKBENCH = new BlockBase("workbench", Material.IRON, 3.0f, 5.0f, "pickaxe", 1, MainRegistry.otherTab, false, false, new AxisAlignedBB(-0.45D, 0, 0, 1.55D, 1.0D, 1.0D), new AxisAlignedBB(0, 0, -0.45D, 1.0D, 1.0D, 1.55D),true);
```
Comment (JustDesoroxxx) [inline src/main/java/cc/panada/mwc/registry/MainRegistry.java:52]: `1.0D` Should just be `1`

**Review by JustDesoroxxx — CHANGES_REQUESTED** [review]

**Comment by SticksDeveloper** [comment]
Actually Resolved this time

File: src/main/java/cc/panada/mwc/registry/MainRegistry.java [hunk 12]
```diff
@@ -71,4 +71,4 @@ public static void preInit() {
             ModelLoader.setCustomModelResourceLocation(item, 0, new ModelResourceLocation(item.getRegistryName(), "inventory"));
         }
     }
-}
+}
```
Comment (JustDesoroxxx) [inline src/main/java/cc/panada/mwc/registry/MainRegistry.java:74]: Missing newline

File: src/main/java/cc/panada/mwc/other/BlockBase.java [hunk 13]
```diff
@@ -123,4 +123,4 @@ public AxisAlignedBB getBoundingBox(IBlockState state, IBlockAccess source, Bloc
         }
     }
 
-}
+}
```
Comment (JustDesoroxxx) [inline src/main/java/cc/panada/mwc/other/BlockBase.java:125]: Missing newline

**Review by JustDesoroxxx — CHANGES_REQUESTED** [review]

**Pull request merged**